src/app/store/effects/app-form.effects.spec.ts

```diff
@@ -48,7 +49,74 @@ describe('AppFormEffects', () => {
 
       const expected$ = cold('--');
 
-      expect(effects.setAppFormPristine$).toBeObservable(expected$);
+      expect(effects.setTechFormPristine$).toBeObservable(expected$);
+    });
+  });
+
+  describe('setTechFormDirty$', () => {
+    it('should dispatch setAppFormDirty action if view state is set to edit or create', () => {
+      action = new SetViewState(VIEW_STATE.EDIT);
+      actions$ = hot('-a--', { a: action });
+
+      const appFormDirty = new SetAppFormDirty();
+
+      const expected$ = cold('-(b)', { b: appFormDirty });
+
+      expect(effects.setTechFormDirty$).toBeObservable(expected$);
+    });
+
+    it('should return an empty stream if view state is set to view only', () => {
+      action = new SetViewState(VIEW_STATE.VIEW_ONLY);
+      actions$ = hot('-a--', { a: action });
+
+      const expected$ = cold('--');
+
+      expect(effects.setTechFormDirty$).toBeObservable(expected$);
+    });
+  });
+
+  describe('setTestFormPristine$', () => {
+    it('should dispatch SetAppFormPristine action if test view state is set to view only', () => {
```

Comment:
similar here use `should dispatch SetAppFormPristine & ClearErrorMessage actions if test view state is set to view only`